packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/DocumentController.java

```diff
@@ -128,6 +143,31 @@ public ResponseEntity<Integer> uploadDocumentFromGithub(
 
 	}
 
+	@PutMapping(value = "/{id}/uploadDocumentFromDOI")
+	public ResponseEntity<Integer> uploadDocumentFromDOI(
+		@PathVariable("id") String id,
+		@RequestParam(name = "doi", required = true) String doi,
+		@RequestParam(name = "filename", required = true) String filename
+	) throws IOException, URISyntaxException {
+			CloseableHttpClient httpclient = HttpClients.custom()
+				.disableRedirectHandling()
+				.build();	
+
+			byte[] fileAsBytes = downloadService.getPDF("https://unpaywall.org/" + doi);
+			HttpEntity fileEntity = new ByteArrayEntity(fileAsBytes, ContentType.APPLICATION_OCTET_STREAM);
+			final PresignedURL presignedURL = proxy.getUploadUrl(id, filename).getBody();
+			final HttpPut put = new HttpPut(presignedURL.getUrl());
+			put.setEntity(fileEntity);
+			final HttpResponse response = httpclient.execute(put);
+
+			if(response.getStatusLine().getStatusCode() < 400) {
+				// fire and forgot pdf extractions
+				knowledgeMiddlewareProxy.postPDFToCosmos(id);
```

Comment:
Please send the `job_id` back to the front-end to handle notification of job being done.

Reply:
@YohannParis 

Example payload on handled errors, a case where the pdf cannot be uploaded:
<img width="637" alt="Screenshot 2023-10-17 at 2 47 09 PM" src="https://github.com/DARPA-ASKEM/terarium/assets/33158416/c86929eb-8cec-480a-9bc6-79a036750529">

Example payload on full success:
<img width="655" alt="Screenshot 2023-10-17 at 2 47 20 PM" src="https://github.com/DARPA-ASKEM/terarium/assets/33158416/1c477f2a-0331-4ba6-9cc5-478b57b0c1a3">

Reply:
Magnificent !!!!